### PR TITLE
Feature/mttl 2076

### PR DESCRIPTION
**Pull request by mikemoody-amido**
## Link to JIRA ticket

https://hackney.atlassian.net/browse/MTTL-2076

## Describe this PR

Create common package with code used in testing Sns events

#### _Checklist_

- [x] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [x] Checked all code for possible refactoring
- [x] Code pipeline builds correctly

### CODE REVIEW

File: Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+This package is designed to assist with testing services that use Sns.
+It contains a fixture that can be used to set up a local Sns instance so that it will be used by code under test.
+
+* [Configuring the Sns fixture](#Configuring-the-Sns-fixture)
+* [ISnsEventVerifier](#ISnsEventVerifier)
+* [ISnsFixture](#ISnsFixture)
+
+## Configuring the Sns fixture
+As a concept, the fixture is a class that ensures that a local Sns instance is set up and accessibly by both the
```
Comment (humulla) [inline Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md]:
```suggestion
As a concept, the fixture is a class that ensures that a local Sns instance is set up and accessible by both the 
```

File: Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+* [ISnsEventVerifier](#ISnsEventVerifier)
+* [ISnsFixture](#ISnsFixture)
+
+## Configuring the Sns fixture
+As a concept, the fixture is a class that ensures that a local Sns instance is set up and accessibly by both the 
+code under test and the tests themselves.
+
+The test project may well contain such a class if there is existing local code that configures a local database instance, 
+however with this common implementation, the order of creation of the application factory and fixture has been inverted. 
+I.e. Here it is the application factory that is set on the xunit collection, and then the factory constructs and
```
Comment (humulla) [inline Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md:16]:
```suggestion
I.e. Here, it is the application factory that is set on the xunit collection, and then the factory constructs and 
```

File: Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+however with this common implementation, the order of creation of the application factory and fixture has been inverted. 
+I.e. Here it is the application factory that is set on the xunit collection, and then the factory constructs and 
+configures the fixture when xunit creates it.
+
+In addition the mock (web) applicaiton factory should implement the Dispose pattern and dispose of the SnsFixture appropriately. 
+In this way all topics and queues created for a test run are removed when it is done.
+
+### Api projects
+#### 1. Create / modify the mock web application factory
+`CreateSnsTopic` should be called oince for each topic that needs to be created.
```
Comment (humulla) [inline Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md]:
```suggestion
`CreateSnsTopic` should be called once for each topic that needs to be created.
```

File: Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        // to be the place to apply [CollectionDefinition] and all the
+        // ICollectionFixture<> interfaces.
+    }
+}
+```
+
+### Non-Api projects
+Non-Api projects construct the `IHostBuilder` slightly differently to that for Api projects.
+#### 1. Create / modify the mock application factory
+`CreateSnsTopic` should be called oince for each topic that needs to be created.
```
Comment (humulla) [inline Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md]:
```suggestion
`CreateSnsTopic` should be called once for each topic that needs to be created.
```

File: Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+                .Then(t => _steps.ThenTheThingCreatedEventIsRaised(_snsFixture))
+                .BDDfy();
+        }
+    }
+}
+```
+
+#### Verifying an event was raised
+Create a method that calls the appropriate `SnsEventVerifier` passing it an action that will be used to test all event messages raised
+to see if it the one expected.
```
Comment (humulla) [inline Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.Sns/README.md]:
```suggestion
to see if it is the one expected.
```

File: README.md [hunk 6]
```diff
@@ -44,7 +44,9 @@ The following features are implemented within this package.
 * [Validation](/Hackney.Core/Hackney.Core.Validation/README.md)
 * [Validation.AspNet](/Hackney.Core/Hackney.Core.Validation.AspNet/ReadMe.md)
 * Testing
+  * [Hackney.Core.Testing.DynamoDb](/Hackney.Core/Hackney.Core.Testing/Hackney.Core.Testing.DynamoDb/README.md)
```
Comment (humulla) [inline README.md:47]: Has the Dynamodb read me been converted to have it own read me (don’t see it in this PR…)

Comment (mikemoody-amido) [reply README.md:47]: No. This is the DynamoDb _testing_ package documentation, which did already exist but which didn't have a link in the root level readme.

**Pull request merged**